### PR TITLE
Add boto_sns module and state.

### PR DESCRIPTION
**Pull request by zimathias**
Include integration tests.

### CODE REVIEW

File: salt/modules/boto_sns.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+def __virtual__():
+    '''
+    Only load if boto libraries exist.
+    '''
+    if not HAS_BOTO:
+        return False
+    return True
+
+
+def get_all_topics(region=None, key=None, keyid=None, profile=None):
```
Comment (thatch45) [inline salt/modules/boto_sns.py]: This function needs a docstring

**Comment by thatch45** [comment]
Thanks @zimathias ! Looks like the only thing we need to add is a docstring for that function, everything else looks fantastic!

File: salt/modules/boto_sns.py (hunk 1 above)
Comment (rallytime) [inline salt/modules/boto_sns.py]: Oh! And a CLI example, too. :)

File: salt/modules/boto_sns.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        salt myminion boto_sns.delete mytopic region=us-east-1
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    conn.delete_topic(get_arn(name, region, key, keyid, profile))
+    log.info('Deleted SNS topic {0}'.format(name))
+    return True
+
+
+def get_arn(name, region=None, key=None, keyid=None, profile=None):
```
Comment (rallytime) [inline salt/modules/boto_sns.py]: This function also needs a docstring with an accompanying CLI example.

**Comment by rallytime** [comment]
Also, thanks for writing tests to go with your additions. That is very much appreciated!

**Comment by zimathias** [comment]
I've documented the functions.

**Comment by ryan-lane** [comment]
Can this be backported to 2015.2? It's a new module add and would be nice for the release :).

**Comment by techhat** [comment]
Ping @rallytime.

**Comment by rallytime** [comment]
Done!

**Pull request merged**